nncf/tensorflow/accuracy_aware_training/keras_model_utils.py

```diff
@@ -62,6 +63,14 @@ def train_epoch_fn(compression_ctrl, model, epoch):
                 steps=data_handler.inferred_steps
             )
 
+        # Check if callbacks.on_train_begin() has been called before the first epoch.
```

Comment:
The only callback which calls at `on_train_begin()` is `History`. It creates an `epoch` attribute, which is used at `on_epoch_end()`. 

Probably, should be a better way to handle it. Do not know how to call  `on_train_begin()` from outside of `train_epoch_fn()` in current design.
@alexsu52 do you have some options in mind?